weatherAplication/Modules/Home/ViewModel/HomeViewModel.swift

```diff
@@ -2,21 +2,44 @@ import Foundation
 
 protocol HomeViewModelContract {
     var delegate: HomeViewModelDelegate? { get set }
+    var lastSearches: [CityInfo] { get set }
 
+    func viewWillAppear()
     func proceedButtonPressed()
+    func cellPressed(cityInfo: CityInfo)
 }
 
 class HomeViewModel: HomeViewModelContract {
 
     var delegate: HomeViewModelDelegate?
+    var lastSearches: [CityInfo] = []
+    let recentCities = StorageService.sharedInstance
+
+    init() {
+        viewWillAppear()
+    }
```

Comment:
Tak w sumie to tego nie potrzebujesz, viewWillAppear samo się wywoła z viewControllera. Na ten moment wywołuje się 2 razy na starcie- raz tutaj w inicie i raz jak widok się pojawia

Reply:
oj tak, funkcja viewWillAppear powinna być wołana jedynie z poziomu controllera, tak przynajmniej oznacza jej nazwa